### PR TITLE
[CHK 2309] feat added get all payment method

### PR DESCRIPTION
**Pull request by Fernando-Granato**
### List of changes

- Added the endpoint for retrieving all payment methods into the openapi **_ecommerce-for-io_**.
          The header x-client-Id io is automatically set in the _base_policy of the api group 👉🏻 https://github.com/pagopa/pagopa-infra/blob/5c799513a1049b4eb5d16c149b987b077651adc7/src/domains/ecommerce-app/api/ecommerce-io/v1/_base_policy.xml.tpl#L37
- Furthermore, the payment-method and checkout open APIs have been updated with the addition of the enum PaymentMethodManagementType in the **PaymentMethodResponse** response and PaymentMethodRequest DTO

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/domains/ecommerce-app/api/ecommerce-checkout/v1/_openapi.json.tpl [hunk 1]
```diff
@@ -1792,6 +1792,10 @@
             "items": {
               "$ref": "#/components/schemas/Range"
             }
+          },
+          "isOnboardable": {
```
Comment (pietro-tota) [inline src/domains/ecommerce-app/api/ecommerce-checkout/v1/_openapi.json.tpl:1796]: I would made this field mandatory in payment method response

Comment (Fernando-Granato) [reply src/domains/ecommerce-app/api/ecommerce-checkout/v1/_openapi.json.tpl:1796]: And if not present on the collection we set it to false by default ?

Comment (pietro-tota) [reply src/domains/ecommerce-app/api/ecommerce-checkout/v1/_openapi.json.tpl:1796]: Yes, I would handle the field absence into payment method logic (or make isOnboardable field mandatory for all payment method documents)

Comment (Fernando-Granato) [reply src/domains/ecommerce-app/api/ecommerce-checkout/v1/_openapi.json.tpl:1796]: Fixed with 09f90ad8f46c088662fd63be94ccea3e85947e3f

**Review by pietro-tota — APPROVED** [review]
🚀

**Review by pietro-tota — APPROVED** [review]
🚀

File: src/domains/ecommerce-app/api/ecommerce-checkout/v1/_openapi.json.tpl [hunk 2]
```diff
@@ -1800,7 +1803,8 @@
           "description",
           "status",
           "paymentTypeCode",
-          "ranges"
+          "ranges",
+          "methodManagement"
```
Comment (infantesimone) [inline src/domains/ecommerce-app/api/ecommerce-checkout/v1/_openapi.json.tpl:1807]: a required field for the api currently used in prod by checkout, is this PR backwards compatible? does it require a definition of a release strategy?

**Review by infantesimone — CHANGES_REQUESTED** [review]

File: src/domains/ecommerce-app/api/ecommerce-checkout/v1/_openapi.json.tpl (hunk 2 above)
Comment (Fernando-Granato) [reply src/domains/ecommerce-app/api/ecommerce-checkout/v1/_openapi.json.tpl:1807]: Momentarily we could avoid updating the openapi of **ecommerce for checkout** by ensuring backward compatibility. What do you think ?

**Review by infantesimone — APPROVED** [review]

**Pull request merged**